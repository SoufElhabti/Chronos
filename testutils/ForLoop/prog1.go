package main

import (
	"fmt"
	"sync"
)

func main() {
	m := make(map[string]string)
	m["2"] = "b" // Second conflicting access.
	for k, v := range m {
		fmt.Println(k, v)
		mutex := sync.Mutex{}
		mutex.Lock()
		m["2"] = "b" // Second conflicting access.
	}
}
