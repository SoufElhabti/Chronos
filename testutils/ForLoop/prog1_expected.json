	{
		"Lockset": {
			"ExistingLocks": {},
			"ExistingUnlocks": {}
		},
		"GuardedAccess": [
			{
				"ID": 1,
				"Pos": 2961365,
				"Value": 2961343,
				"OpKind": 1,
				"LocksetJson": {
					"ExistingLocks": {},
					"ExistingUnlocks": {}
				},
				"State": {
					"GoroutineID": 1,
					"Clock": {
						"1": 1
					}
				}
			},
			{
				"ID": 2,
				"Pos": 2961365,
				"Value": 0,
				"OpKind": 0,
				"LocksetJson": {
					"ExistingLocks": {},
					"ExistingUnlocks": {}
				},
				"State": {
					"GoroutineID": 1,
					"Clock": {
						"1": 2
					}
				}
			},
			{
				"ID": 3,
				"Pos": 2961408,
				"Value": 2961343,
				"OpKind": 0,
				"LocksetJson": {
					"ExistingLocks": {},
					"ExistingUnlocks": {}
				},
				"State": {
					"GoroutineID": 1,
					"Clock": {
						"1": 3
					}
				}
			},
			{
				"ID": 5,
				"Pos": 2961432,
				"Value": 2961343,
				"OpKind": 0,
				"LocksetJson": {
					"ExistingLocks": {},
					"ExistingUnlocks": {}
				},
				"State": {
					"GoroutineID": 1,
					"Clock": {
						"1": 5
					}
				}
			},
			{
				"ID": 7,
				"Pos": 0,
				"Value": 2961473,
				"OpKind": 0,
				"LocksetJson": {
					"ExistingLocks": {},
					"ExistingUnlocks": {}
				},
				"State": {
					"GoroutineID": 1,
					"Clock": {
						"1": 7
					}
				}
			},
			{
				"ID": 8,
				"Pos": 2961469,
				"Value": 0,
				"OpKind": 0,
				"LocksetJson": {
					"ExistingLocks": {},
					"ExistingUnlocks": {}
				},
				"State": {
					"GoroutineID": 1,
					"Clock": {
						"1": 8
					}
				}
			},
			{
				"ID": 9,
				"Pos": 2961469,
				"Value": 0,
				"OpKind": 1,
				"LocksetJson": {
					"ExistingLocks": {},
					"ExistingUnlocks": {}
				},
				"State": {
					"GoroutineID": 1,
					"Clock": {
						"1": 9
					}
				}
			},
			{
				"ID": 10,
				"Pos": 0,
				"Value": 2961473,
				"OpKind": 0,
				"LocksetJson": {
					"ExistingLocks": {},
					"ExistingUnlocks": {}
				},
				"State": {
					"GoroutineID": 1,
					"Clock": {
						"1": 10
					}
				}
			},
			{
				"ID": 11,
				"Pos": 2961472,
				"Value": 0,
				"OpKind": 0,
				"LocksetJson": {
					"ExistingLocks": {},
					"ExistingUnlocks": {}
				},
				"State": {
					"GoroutineID": 1,
					"Clock": {
						"1": 11
					}
				}
			},
			{
				"ID": 12,
				"Pos": 2961472,
				"Value": 0,
				"OpKind": 1,
				"LocksetJson": {
					"ExistingLocks": {},
					"ExistingUnlocks": {}
				},
				"State": {
					"GoroutineID": 1,
					"Clock": {
						"1": 12
					}
				}
			},
			{
				"ID": 13,
				"Pos": 0,
				"Value": 2961494,
				"OpKind": 0,
				"LocksetJson": {
					"ExistingLocks": {},
					"ExistingUnlocks": {}
				},
				"State": {
					"GoroutineID": 1,
					"Clock": {
						"1": 13
					}
				}
			},
			{
				"ID": 14,
				"Pos": 2961490,
				"Value": 0,
				"OpKind": 0,
				"LocksetJson": {
					"ExistingLocks": {},
					"ExistingUnlocks": {}
				},
				"State": {
					"GoroutineID": 1,
					"Clock": {
						"1": 14
					}
				}
			},
			{
				"ID": 15,
				"Pos": 2961490,
				"Value": 0,
				"OpKind": 1,
				"LocksetJson": {
					"ExistingLocks": {},
					"ExistingUnlocks": {}
				},
				"State": {
					"GoroutineID": 1,
					"Clock": {
						"1": 15
					}
				}
			},
			{
				"ID": 16,
				"Pos": 0,
				"Value": 2961494,
				"OpKind": 0,
				"LocksetJson": {
					"ExistingLocks": {},
					"ExistingUnlocks": {}
				},
				"State": {
					"GoroutineID": 1,
					"Clock": {
						"1": 16
					}
				}
			},
			{
				"ID": 17,
				"Pos": 2961493,
				"Value": 0,
				"OpKind": 0,
				"LocksetJson": {
					"ExistingLocks": {},
					"ExistingUnlocks": {}
				},
				"State": {
					"GoroutineID": 1,
					"Clock": {
						"1": 17
					}
				}
			},
			{
				"ID": 18,
				"Pos": 2961493,
				"Value": 0,
				"OpKind": 1,
				"LocksetJson": {
					"ExistingLocks": {},
					"ExistingUnlocks": {}
				},
				"State": {
					"GoroutineID": 1,
					"Clock": {
						"1": 18
					}
				}
			},
			{
				"ID": 19,
				"Pos": 2961504,
				"Value": 0,
				"OpKind": 0,
				"LocksetJson": {
					"ExistingLocks": {},
					"ExistingUnlocks": {}
				},
				"State": {
					"GoroutineID": 1,
					"Clock": {
						"1": 19
					}
				}
			},
			{
				"ID": 20,
				"Pos": 2961504,
				"Value": 0,
				"OpKind": 0,
				"LocksetJson": {
					"ExistingLocks": {},
					"ExistingUnlocks": {}
				},
				"State": {
					"GoroutineID": 1,
					"Clock": {
						"1": 20
					}
				}
			},
			{
				"ID": 21,
				"Pos": 0,
				"Value": 2961504,
				"OpKind": 0,
				"LocksetJson": {
					"ExistingLocks": {},
					"ExistingUnlocks": {}
				},
				"State": {
					"GoroutineID": 1,
					"Clock": {
						"1": 21
					}
				}
			},
			{
				"ID": 22,
				"Pos": 0,
				"Value": 2961532,
				"OpKind": 0,
				"LocksetJson": {
					"ExistingLocks": {},
					"ExistingUnlocks": {}
				},
				"State": {
					"GoroutineID": 1,
					"Clock": {
						"1": 22
					}
				}
			},
			{
				"ID": 23,
				"Pos": 2961528,
				"Value": 0,
				"OpKind": 0,
				"LocksetJson": {
					"ExistingLocks": {},
					"ExistingUnlocks": {}
				},
				"State": {
					"GoroutineID": 1,
					"Clock": {
						"1": 23
					}
				}
			},
			{
				"ID": 24,
				"Pos": 2961528,
				"Value": 0,
				"OpKind": 1,
				"LocksetJson": {
					"ExistingLocks": {},
					"ExistingUnlocks": {}
				},
				"State": {
					"GoroutineID": 1,
					"Clock": {
						"1": 24
					}
				}
			},
			{
				"ID": 25,
				"Pos": 0,
				"Value": 2961532,
				"OpKind": 0,
				"LocksetJson": {
					"ExistingLocks": {},
					"ExistingUnlocks": {}
				},
				"State": {
					"GoroutineID": 1,
					"Clock": {
						"1": 25
					}
				}
			},
			{
				"ID": 26,
				"Pos": 2961531,
				"Value": 0,
				"OpKind": 0,
				"LocksetJson": {
					"ExistingLocks": {},
					"ExistingUnlocks": {}
				},
				"State": {
					"GoroutineID": 1,
					"Clock": {
						"1": 26
					}
				}
			},
			{
				"ID": 27,
				"Pos": 2961531,
				"Value": 0,
				"OpKind": 1,
				"LocksetJson": {
					"ExistingLocks": {},
					"ExistingUnlocks": {}
				},
				"State": {
					"GoroutineID": 1,
					"Clock": {
						"1": 27
					}
				}
			},
			{
				"ID": 28,
				"Pos": 2961584,
				"Value": 2961343,
				"OpKind": 1,
				"LocksetJson": {
					"ExistingLocks": {
						"t332961542": 2961577
					},
					"ExistingUnlocks": {}
				},
				"State": {
					"GoroutineID": 1,
					"Clock": {
						"1": 28
					}
				}
			},
			{
				"ID": 29,
				"Pos": 2961584,
				"Value": 0,
				"OpKind": 0,
				"LocksetJson": {
					"ExistingLocks": {
						"t332961542": 2961577
					},
					"ExistingUnlocks": {}
				},
				"State": {
					"GoroutineID": 1,
					"Clock": {
						"1": 29
					}
				}
			},
			{
				"ID": 6,
				"Pos": 0,
				"Value": 0,
				"OpKind": 0,
				"LocksetJson": {
					"ExistingLocks": {},
					"ExistingUnlocks": {}
				},
				"State": {
					"GoroutineID": 1,
					"Clock": {
						"1": 6
					}
				}
			},
			{
				"ID": 4,
				"Pos": 0,
				"Value": 0,
				"OpKind": 0,
				"LocksetJson": {
					"ExistingLocks": {},
					"ExistingUnlocks": {}
				},
				"State": {
					"GoroutineID": 1,
					"Clock": {
						"1": 4
					}
				}
			}
		]
	}